R/session_check_reauth.R

```diff
@@ -0,0 +1,61 @@
+#'@title Checks current session and re-authenticates if necessary
+#'
+#'@inheritParams authenticate_sb
+#'@param session (Advanced) Alternatively supply a specific session to be checked
+#'@param ... Any additional parameters are ignored. 
+#'
+#'@description
+#'This checks the state of your Sciencebase session and, if invalid or if you've supplied
+#'a new user name, it will re-authenticate. This is primarily a helper function for 
+#'packages that reverse-depend on sbtools.
+#'
+#'@examples
+#'
+#'\dontrun{
+#'
+#'#will prompt for username/pass if needed
+#'session_check_reauth()
+#'
+#'session_check_reauth('user@@usgs.gov')
+#'
+#'#default behavior
+#'session_check_reauth(session=current_session())
+#'
+#'}
+#'
+#'
+#'@export
+session_check_reauth = function(username, password, session, ...){
```

Comment:
I'm going to leave it out of sbtools. I want an user to specifically choose this functionality as, usually, I don't assume most of my function calls are going to try and create a user dialog. Putting this into the sbtools functions would embed interactive logic very deeply into the package, right now interactive code sits near the surface, in two functions, authenticate_sb and now this new function.